### ISSUE TITLE
[VersionControl] Publish a new project to Git fails

### ISSUE THREAD

**Issue by jsuarezruiz** [mono/monodevelop#7163]
The steps to reproduce the problem:
- Create a new project or open a project not using VCS.
- Choose the option Publis to Version Control... from the Version Control menu.
- Nothing happens (internally an exception).

NOTE: I used a Xamarin.Forms sample (a solution with 3 projects).

```
ERROR [2019-02-15 11:27:45Z]: Version control operation failed: 
System.ArgumentNullException: Value cannot be null.
Parameter name: value
  at System.String.StartsWith (System.String value, System.StringComparison comparisonType) [0x00107] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/String.Comparison.cs:980 
  at MonoDevelop.Core.FilePath.IsChildPathOf (MonoDevelop.Core.FilePath basePath) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/FilePath.cs:179 
  at MonoDevelop.VersionControl.Repository.Publish (System.String serverPath, MonoDevelop.Core.FilePath localPath, MonoDevelop.Core.FilePath[] files, System.String message, MonoDevelop.Core.ProgressMonitor monitor) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/Repository.cs:530 
  at MonoDevelop.VersionControl.PublishCommand+PublishWorker.Run () [0x00002] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/PublishCommand.cs:98 
  at MonoDevelop.VersionControl.VersionControlTask.<BackgroundWorker>b__13_0 () [0x00000] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/VersionControlTask.cs:51 
  at System.Threading.Tasks.Task.InnerInvoke () [0x0000f] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:2500 
  at System.Threading.Tasks.Task.Execute () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:2343 
```

> VS bug [#796046](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/796046)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.1631 (master)

Pull Request #7178 merged by: Vsevolod Kukol
Author: mono
Commit: 2b2864f1c8eb972bf39aefd9561861b3fed65f6c (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2410 (release-8.0)

Pull Request #7178 merged by: Vsevolod Kukol
Author: mono
Commit: 2b2864f1c8eb972bf39aefd9561861b3fed65f6c (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.2.8 (release-7.8)

Author: Javier Suárez Ruiz
Commit: 497763b07ce22c126b8ac41a38c38293481d9072 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #7178 merged by: Vsevolod Kukol
Author: mono
Commit: 2b2864f1c8eb972bf39aefd9561861b3fed65f6c (xamarin/monodevelop)
